src/provisioner/cluster.py

```diff
@@ -87,26 +87,80 @@ def _get_kvs(self, prefix, node):
             kvs.append((prefix, node))
         return kvs
 
-    def _add_component_version(self, components):
-        """Add version for each component."""
-        updated_components = []
-        for component in components:
-            component['version'] = self._cortx_release.get_version(component['name'])
-            updated_components.append(component)
-        return updated_components
+    def _get_component_version(self, component_name):
+        """Return component version."""
+        version = self._cortx_release.get_version(component_name)
+        return version
+
+    def _get_component_key_value_list(self, component_list: list, node_id: str):
+        """Return list of confstore keys-values for components."""
+        # Create confstore keys, from component_list i.e. config from config.yaml.
+        component_key_value_list = []
+        for index, component in enumerate(component_list):
+            key_prefix = f'node>{node_id}>components[{index}]'
+            component_name = component['name']
+            version = self._get_component_version(component_name)
+            component_key_value_list.extend((
+                (f'{key_prefix}>name', component_name),
+                (f'{key_prefix}>version', version)))
+
+            service_list = component.get('services')
+            if service_list:
+                for index, service in enumerate(service_list):
+                    component_key_value_list.append((
+                        f'{key_prefix}>services[{index}]', service))
+
+        return component_key_value_list
+
+    def _get_cvg_key_value_list(self, cvg_list: dict, node_id: str):
```

Comment:
suggestion: _get_storage_attrs() or _get_storage_kv_list()

Reply:
Renamed to _get_storage_kv_list().